### DISCUSSION TITLE
Manipulate request authentication

### DISCUSSION THREAD

**Discussion by araujo-filipe** [AbsaOSS/spline-spark-agent#215] [Q&A] [answered]
Hey,

Is it possible to manipulate the spline agent request somehow to add headers or query parameters?

Use case is I'm trying to apply some different security to the REST gateway which needs requests to be "signed" either by adding a new header or a query parameter. Is that possible out of the box using some property or something?

**Comment by cerveada** [comment, marked as answer]
Hello,
not by property, but it should be fairly easy to create your own dispatcher.

Look at `DefaultSplineConfigurer` there is a property to set your own dispatcher class instead of the default one.

You can basically reuse the `HttpLineageDispatcher` and just modify `RestClient` and mainly`RestEndpoint`.
Then just set the path to your modified dispatcher via the property and the agent will use it.
